### PR TITLE
Update golangci-lint

### PR DESCRIPTION
**Pull request by SuperQ**
* Update golangci-lint to v1.53.3.
* Update the sync script handler for the old golanci-lint action.

### CODE REVIEW

**Comment by roidelapluie** [comment]
yay unused-parameter

**Review by LeviHarrison — APPROVED** [review]
LGTM, although we may want to disable `unused-parameter` (or not)

**Comment by roidelapluie** [comment]
linting failure

**Comment by SuperQ** [comment]
Disabled unused-parameter.

File: scripts/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+# This action is synced from https://github.com/prometheus/prometheus
+name: golangci-lint
+on:
```
Comment (mmorel-35) [inline scripts/golangci-lint.yml:4]: I have created #12235 to fix this

**Comment by roidelapluie** [comment]
cc @SuperQ do you want to rebase this? Thanks!

**Comment by SuperQ** [comment]
Rebase done, updated to latest version.

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**